File: Data Structures & Algorithms/count-paths/submission-5.cpp
class Solution {
public:
    // int solve(int x, int y, int m, int n, vector<vector<int>> &dp) {
    //     if (x > m-1 || y > n-1) return 0;
    //     if (x == m-1 && y == n-1) {
    //         return 1;
    //     }
    //     if (dp[x][y] != -1) return dp[x][y];
    //     dp[x][y] = solve(x+1, y, m, n, dp) + solve(x, y+1, m, n, dp);
    //     return dp[x][y];
    // }
    int uniquePaths(int m, int n) {
        vector<vector<int>> dp(m, vector<int>(n,1));
        for (int i = 1; i < m; i++) {
            for (int j = 1; j < n; j++) {
                dp[i][j] = dp[i][j-1] + dp[i-1][j];
            }
        }
        return dp[m-1][n-1];
    }
};

File: Data Structures & Algorithms/surrounded-regions/submission-5.py
class Solution:
    def solve(self, board: List[List[str]]) -> None:
        q = deque()
        n, m = len(board), len(board[0])
        vis = [[0 for _ in range(m)] for _ in range(n)]
        for i in range(m):
            if board[0][i] == 'O' and not vis[0][i]:
                q.append((0,i))
                vis[0][i] = 1
            if board[n-1][i] == 'O' and not vis[n-1][i]:
                q.append((n-1,i))
                vis[n-1][i] = 1
        for i in range(1,n-1):
            if board[i][0] == 'O' and not vis[i][0]:
                q.append((i,0))
                vis[i][0] = 1
            if board[i][m-1] == 'O' and not vis[i][m-1]:
                q.append((i,m-1))
                vis[i][m-1] = 1
        delr = [-1, 0, 1, 0]
        delc = [0, -1, 0, 1]
        while q:
            x, y = q.popleft()
            for i in range(4):
                nr = x + delr[i]
                nc = y + delc[i]
                if 0 <= nr < n and 0 <= nc < m and board[nr][nc] == 'O' and not vis[nr][nc]:
                    q.append((nr,nc))
                    vis[nr][nc] = 1
        for i in range(n):
            for j in range(m):
                if board[i][j] == 'O' and not vis[i][j]:
                    board[i][j] = 'X'